src/ReinforcementLearningCore/src/core/run.jl

```diff
@@ -62,21 +62,20 @@ Base.show(io::IO, m::MIME"text/plain", t::Experiment{S}) where {S} = show(io, m,
 Base.run(ex::Experiment) = run(ex.policy, ex.env, ex.stop_condition, ex.hook)
 
 function Base.run(
-    policy::AbstractPolicy,
-    env::AbstractEnv,
+    policy::P,
+    env::E,
     stop_condition=StopAfterEpisode(1),
     hook=EmptyHook(),
     reset_condition=ResetAtTerminal()
-)
+) where {P<:AbstractPolicy, E<:AbstractEnv}
```

Comment:
Why this? They should be strictly equivalent if I'm not mistaken. If there is no reason, I'd prefer we keep the old syntax as it is the canonical Julia syntax (we don't need E or P in the namespace).